### ISSUE TITLE
[docs][pickers] Add a11y/keyboard navigation documentation

### ISSUE THREAD

**Issue by LukasTy** [mui/mui-x#6451]
Add `Date and Time pickers` accessibility documentation page.

Inspiration can be taken from [`Data Grid` page](https://mui.com/x/react-data-grid/accessibility).

Special attention should be taken to keyboard navigation—new `fields` components will add a significant amount of custom behaviours.
